### PR TITLE
Create a database based on a diagram

### PR DESCRIPTION
**Pull request by K0ppai**
Dear Reviewers,
We've worked on the following in this pull request,

1. Create `schema_based_on_diagram.sql` file.
2. Create tables based on the diagram.
3. Create Join tables for many-to-many relationships.
4. Add the FK indexes.

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+);
+
+CREATE TABLE invoices(
+	id INT GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
+	total_amount DECIMAL(8,2),
+	generated_at TIMESTAMP,
+	payed_at TIMESTAMP,
+	medical_history_id INT REFERENCES medical_histories(id)
+);
+
```
Comment (gemmen29) [inline schema_based_on_diagram.sql:41]: - [ ] Kindly create indexes for all foreign keys as per this [requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/database_from_diagram.md#:~:text=Remember%20to%20add%20the%20FK%20indexes.).

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram.sql [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+CREATE TABLE treatments(
+	id INT GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
+	type VARCHAR(250),
+	name VARCHAR(250)
+);
+
+CREATE TABLE medical_histories_and_treatments(
+	treatment_id INT REFERENCES treatments(id),
+	medical_histories_id INT REFERENCES medical_histories(id)
+);
```
Comment (AbelG101) [inline schema_based_on_diagram.sql]: - [ ] Ouch! this is close but you should also add indexes for the foreign keys found in the join table highlighted above as they're also considered foreign keys. 😉

**Review by AbelG101 — CHANGES_REQUESTED** [review]
# Hi @K0ppai  & @tonnytech 👋, 

You're doing a good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- All tables are properly created from the diagram ✔
- Primary keys and foreign keys are correctly set ✔
- Correct use of Git Flow ✔
- Descriptive PR title with a great summary ✔

## Required Changes ♻️
- _Check the comments under the review._
- you just have to make a few changes and you're on your way 🚀.


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ShadyShawkat — APPROVED** [review]
## Hi @K0ppai, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 
- Your readme looks professional ✔️ 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by K0ppai** [comment]
Thanks, reviewers @gemmen29, @ShadyShawkat, @AbelG101 for the reviews.

**Comment by K0ppai** [comment]
Thanks, reviewers for the reviews. @gemmen29 , @ShadyShawkat , @AbelG101

**Pull request closed without merging**